monai/metrics/rocauc.py

```diff
@@ -66,9 +66,17 @@ def aggregate(self):  # type: ignore
 def _calculate(y_pred: torch.Tensor, y: torch.Tensor) -> float:
     if not (y.ndimension() == y_pred.ndimension() == 1 and len(y) == len(y_pred)):
         raise AssertionError("y and y_pred must be 1 dimension data with same length.")
-    if not y.unique().equal(torch.tensor([0, 1], dtype=y.dtype, device=y.device)):
-        warnings.warn("y values must be 0 or 1, can not be all 0 or all 1, skip AUC computation and return `Nan`.")
+    y_unique = y.unique()
+    if y_unique.equal(torch.tensor([0], dtype=y.dtype, device=y.device)):
+        warnings.warn("y values can not be all 0, skip AUC computation and return `Nan`.")
         return float("nan")
+    if y_unique.equal(torch.tensor([1], dtype=y.dtype, device=y.device)):
+        warnings.warn("y values can not be all 1, skip AUC computation and return `Nan`.")
+        return float("nan")
+    if not y_unique.equal(torch.tensor([0, 1], dtype=y.dtype, device=y.device)):
+        warnings.warn("y values must be 0 or 1, skip AUC computation and return `Nan`.")
```

Comment:
perhaps just print `y_unique` in the message instead of 3 comparisons?

Reply:
Hi @wyli ,

Thanks for your suggestion, the problem is that: if users forgot to convert labels to [0, 1], it may have many many unique values.
How about change it to 2 prints: (1) all 0 or all 1, (2) not only [0, 1] ?

BTW, I think the sklearn solution you shared also have 3 separated errors for this check:
https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/metrics/_ranking.py#L991
https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/metrics/_ranking.py#L1000
https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/metrics/_ranking.py#L549

Thanks.

Reply:
I think if `y_unique` has many elements we can only print first a few?

Reply:
Hi @wyli ,

I simplified the logic in the latest commit, could you please help review it again?
Some test result:
```
y values can not be all 1.0, skip AUC computation and return `Nan`.
y values can not be all 0.0, skip AUC computation and return `Nan`.
..y values must be 0 or 1, but in [0.0, 1.0, 2.0, 3.0], skip AUC computation and return `Nan`.
```

Thanks in advance.